### PR TITLE
[POC] Playing around with compute at

### PR DESCRIPTION
**Pull request by csarofeen**
Playing around with some compute at modes. If compute at doesn't work the standard mode will throw an assertion. The compute at best effort will fail gracefully, so if we can't computeAt a particular tensor, it will fall back to a position that we can compute at. Compute at inline most will try to inline as much as possible all the tensors.

See softmax 3d test case that's modified

### CODE REVIEW

**Comment by csarofeen** [comment]
Closed in favor of: https://github.com/csarofeen/pytorch/pull/687
